### PR TITLE
Add support for user/password auth - Elasticsearch

### PR DESCRIPTION
**Pull request by v-jizhang**
Cherry-pick of
https://github.com/trinodb/trino/pull/2591/commits/bd4b3dd0b77ece2d56d58a38834f952e3f970905

Co-authored-by: Martin Traverso <mtraverso@gmail.com>

Test plan - (Please fill in how you tested your changes)

```
== RELEASE NOTES ==

ElasticSearch Connector Changes
* Add support for Elasticsearch user and password authentication.
   :issue:`15909`
```

### CODE REVIEW

**Comment by abmn614** [comment]
Why hasn't this PR been submitted yet? anyone knows any news？

**Comment by abmn614** [comment]
Why hasn't this PR been submitted yet? Anyone knows some news?

**Comment by v-jizhang** [comment]
Reviewed in #15877

**Comment by zhenxiao** [comment]
hi @v-jizhang could you please resolved the conflict?

**Review by rschlussel — COMMENTED** [review]
I'll leave @zhenxiao to do a proper review, but just a couple things I noticed.

File: presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/ElasticsearchQueryRunner.java [hunk 1]
```diff
@@ -120,6 +120,9 @@ public static Session createSession()
     public static void main(String[] args)
             throws Exception
     {
+        // To start Elasticsearch:
```
[the remark this answers is not in the document]
Comment (v-jizhang) [reply presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/ElasticsearchQueryRunner.java:132]: Working on it.

File: presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConfig.java [hunk 2]
```diff
@@ -32,7 +32,8 @@
 {
     public enum Security
     {
-        AWS
+        AWS,
+        PASSWORD,
```
Comment (zhenxiao) [inline presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConfig.java:36]: should we remove the `,` after `PASSWORD`?

File: presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/TestElasticsearchIntegrationSmokeTest.java [hunk 3]
```diff
@@ -57,12 +57,15 @@
     protected QueryRunner createQueryRunner()
             throws Exception
     {
-        elasticsearch = new ElasticsearchServer();
+        elasticsearch = new ElasticsearchServer("docker.elastic.co/elasticsearch/elasticsearch-oss:6.0.0", ImmutableMap.of());
```
Comment (zhenxiao) [inline presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/TestElasticsearchIntegrationSmokeTest.java:60]: shall we make it a constant or configurable?
`docker.elastic.co/elasticsearch/elasticsearch-oss:6.0.0`

**Review by zhenxiao — APPROVED** [review]
hi @v-jizhang nice work
looks good. 2 minor things

File: presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/TestElasticsearchIntegrationSmokeTest.java (hunk 3 above)
Comment (zhenxiao) [reply presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/TestElasticsearchIntegrationSmokeTest.java:60]: kindly ping

Comment (v-jizhang) [reply presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/TestElasticsearchIntegrationSmokeTest.java]: Made it a constant. Thanks for your reviews.

File: presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConfig.java (hunk 2 above)
Comment (v-jizhang) [reply presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConfig.java]: Removed.

**Comment by zhenxiao** [comment]
looks good. @v-jizhang could you please take a look at the 2 failing tests?

**Comment by zhenxiao** [comment]
nice. thank you, @v-jizhang could you please merge the last commit into previous ones?
I plan to merge this PR after the change

**Pull request merged**